text/0000-fabric-x-next-generation.md

```diff
@@ -0,0 +1,255 @@
+    * **Sidecar Service:** Acts as a dedicated communication bridge. It sits between the ordering service (Arma) and the Coordinator Service. Its responsibilities include fetching blocks from Arma, forwarding them to the Coordinator, managing the local block store, receiving consolidated transaction statuses from the Coordinator, and notifying registered clients about transaction events.
+    * **Coordinator Service:** Orchestrates the validation and commit phases. It receives blocks from the Sidecar, builds the dependency graph, and dispatches transactions to Signature Verifiers and then to Validator-Committer services.
+    * **Signature Verifier Service:** Dedicated to verifying endorsement signatures against policies.
+    * **Validator-Committer (VC) Service:** Responsible for ledger state validation (read-set validation) and committing valid transactions to the distributed database.
+    * **Query Service:** Handles read-only ledger queries efficiently.
+* **Decomposed Orderer (Arma):** The "order" phase is managed by Arma, a microservice-based ordering service:
+    * **Router:** Validates incoming transactions and routes them to appropriate shards.
+    * **Batcher:** Bundles transactions into batches within shards.
+    * **Consenter:** Participates in the BFT consensus protocol on batch digests.
+    * **Assembler:** Reconstructs full blocks from ordered batch digests and batches.
+* **Sharded Distributed Database:** The ledger state is managed by a sharded, strongly consistent distributed database, allowing for greater concurrency and overcoming single-node storage/IO limitations. Developers need to be aware that data is distributed, though the interaction model via chaincode during endorsement should remain largely consistent for basic operations.
+* **Programming Model:** Fabric-X enhances the programming model by allowing application development beyond traditional chaincode-centric architectures while still benefiting from the security through endorsement policies.
+* **Threshold Signatures & Traditional Policies with Pre-stored Identities:** Fabric-X will support threshold signatures for compact, efficient validation. Additionally, traditional endorsement policies will be supported, with an optimization to pre-store identities (e.g., X.509 certificate details) in a configuration block, allowing transactions to reference these identities rather than including full certificates with each signature, thus reducing transaction size.
+* **Dependency Analysis Enables Parallel Verification and Validation:** The Coordinator constructs a dependency graph for transactions. This allows transactions without conflicts to be verified and validated in parallel by multiple signature-verifier and VC instances, significantly speeding up the commit phase.
+* **Flattened Transaction Structure:** To reduce processing overhead associated with serialization/deserialization of deeply nested structures, Fabric-X will adopt a flatter transaction model.
```

Comment:
What is the exact transaction format and structure in Fabric-X, and how does it differ from current Fabric transactions?